torch/csrc/api/include/torch/nn/options/instancenorm.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <torch/arg.h>
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/types.h>
+
+namespace torch {
+namespace nn {
+
+struct TORCH_API InstanceNormOptions {
+  /* implicit */ InstanceNormOptions(int64_t num_features);
+  TORCH_ARG(int64_t, num_features);
```

Comment:
Since `InstanceNormOptions` is used in the functional form and it doesn't need the `num_features` parameter, we can make `num_features` an object of `c10::optional<int64_t>` and make its default value `c10::nullopt`, and add a default constructor for `InstanceNormOptions` so that the functional form's signature can be
```cpp
inline Tensor instance_norm(const Tensor& input, const InstanceNormOptions& options = {}) {
```